src/runtime_src/core/tools/xbmgmt2/OO_UpdateBase.cpp

```diff
@@ -0,0 +1,755 @@
+}
+
+static std::string
+getBDF(unsigned int index)
+{
+  auto dev = xrt_core::get_mgmtpf_device(index);
+  auto bdf = xrt_core::device_query<xrt_core::query::pcie_bdf>(dev);
+  return xrt_core::query::pcie_bdf::to_string(bdf);
+}
+
+static bool
+is_SC_fixed(unsigned int index)
+{
+  try {
+    auto device = xrt_core::get_mgmtpf_device(index);
+    return xrt_core::device_query<xrt_core::query::is_sc_fixed>(device);
+  } catch (...) {
+    // TODO Catching all the exceptions for now. We may need to catch specific exceptions
+    // Work-around. Assume that sc is not fixed if above query throws an exception
+    return false;
+  }
+}
+
+// versal flow to flash sc
+static void
+update_versal_SC(std::shared_ptr<xrt_core::device> dev)
+{
+  std::thread t;
+  std::atomic<bool> done(false);
+  std::shared_ptr<XBU::ProgressBar> progress_reporter;
+  try {
+    const uint32_t val = xrt_core::query::program_sc::value_type(1);
+    // timeout for xgq is 300 seconds
+    const unsigned int max_duration = 300;
+    progress_reporter = std::make_shared<XBU::ProgressBar>("Programming SC", 
+                                max_duration, true /*batch mode for dots*/, std::cout); 
+    progress_reporter.get()->setPrintPercentBatch(false);
+    // Print progress while sc is flashed
+    t = std::thread([&progress_reporter, &done]() {
+      unsigned int counter = 0;
+      while ((counter < progress_reporter.get()->getMaxIterations()) && !done) {
+        progress_reporter.get()->update(counter++);
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+      }
+    });
+    xrt_core::device_update<xrt_core::query::program_sc>(dev.get(), val);
+    done = true;
+    progress_reporter.get()->finish(true, "SC firmware image has been programmed successfully.");
+    t.join();
+  } catch (const xrt_core::query::sysfs_error& e) {
+    done = true;
+    progress_reporter.get()->finish(false, "Failed to update SC flash image.");
+    t.join();
+    throw xrt_core::error(std::string("Error accessing sysfs entry : ") + e.what());
+  }
+}
+
+// Update SC firmware on the board
+static void
+update_SC(unsigned int index, const std::string& file)
+{
+  Flasher flasher(index);
+
+  if (!flasher.isValid())
+    throw xrt_core::error(boost::str(boost::format("%d is an invalid index") % index));
+
+  auto dev = xrt_core::get_mgmtpf_device(index);
+
+  bool is_versal = xrt_core::device_query<xrt_core::query::is_versal>(dev);
+  if (is_versal) {
+    update_versal_SC(dev);
+    return;
+  }
+
+  // if factory image, update SC
+  auto is_mfg = xrt_core::device_query<xrt_core::query::is_mfg>(dev);
+  if (is_mfg) {
+    std::unique_ptr<firmwareImage> bmc = std::make_unique<firmwareImage>(file, BMC_FIRMWARE);
+    if (bmc->fail())
+      throw xrt_core::error(boost::str(boost::format("Failed to read %s") % file));
+
+    if (flasher.upgradeBMCFirmware(bmc.get()) != 0)
+      throw xrt_core::error("Failed to update SC flash image");
+    return;
```

Comment:
Just makes it easier to read.
```suggestion

    return;
```